src/datalink/mod.rs

```diff
@@ -229,6 +229,15 @@ impl NetworkInterface {
     }
 }
 
+/// Represents an IP address and subnet mask
+#[derive(Clone, PartialEq, Eq, Debug, Hash)]
+pub struct IpNetmask {
+    /// IP address
+    pub ip: IpAddr,
+    /// Subnet mask
+    pub netmask: Option<IpAddr>,
```

Comment:
Is there ever an occasion where we have an IP, but don't have a netmask?

If there is "no netmask" (`None`), is that not equivalent to `Ipv4(0,0,0,0)` or `Ipv6(0,0,0,0,0,0,0,0)`?